### ISSUE TITLE
LSTM with multiple inputs and multiple outputs

### ISSUE THREAD

**Issue by ldelosieres** [keras-team/keras#11171]
I am trying to learn the positions of the interesting bits to flip based on a binary sample. To make it simple, I am chunking the binary sample in sequences of 3 bits, i.e., the neural network accepts 3 bits and outputs 3 bits.

Since the binary input can be very long (>64 kbits), I am using a stateful LSTM; however, I am getting poor performances (accuracy around 66%).

I tried LSTM->Dense, LSTM->Dropout->Dense, and LSTM->Dropout->LSTM->Dense, but I am getting similar results.

Please do point out why the algorithm is not converging.

Program:

```
import sys
import copy
import random
import numpy as np
import time
from keras import models
from keras import layers
from keras import optimizers
from keras.utils import to_categorical
import os

K=3 #sequence length

def create_model():
  visible = layers.Input(batch_shape=(1, K, 2))
  cell1 = layers.LSTM(7, return_sequences=True, stateful=True)(visible)
  dropout1 = layers.Dropout(0.05)(cell1)
  dense1 = layers.Dense(2, activation="softmax")(dropout1)

  model = models.Model(inputs=visible, outputs=dense1)
  model.compile(
      optimizer = "adam",
      loss = "binary_crossentropy",
      metrics = ["mae", "acc"]
  )
  return model

X1 = [0, 1, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0]
Y1 = [0, 1, 0, 1, 1, 0, 1, 0, 1, 1, 1, 1]

model = create_model()

#Number of epochs
for j in range(500):
  for i in range(len(X1)/K):
    X = to_categorical(X1[i*K:(i+1)*K], 2)
    Y = to_categorical(Y1[i*K:(i+1)*K], 2)
    ret = model.train_on_batch(np.array([X]), np.array([Y]))
    print(ret)`

  #Once the whole sample is trained, reset the states.
  model.reset_states()
```

**Comment by tomastheod-ITI** [comment]
@ldelosieres A sequence length of 3 is probably too short and you cannot take advantage of the capabilities of LSTM networks. If it makes sense for your problem, try something like 30, 100, or even 300 if you have the resources. Start experimenting with the smaller number and see if you gain anything.

**Comment by ldelosieres** [comment]
@tomastheod-ITI  thank you for your answer. If I choose a K of 12, I have very good results since the problem becomes a “stateless” problem (I can replace stateful=True by stateful=False); however, if K is less than 12, the accuracy is not converging towards 1.0. I increased the X1/Y1 length to 40*12 = 480, and I am getting better results with K set to 30 / 100 / 300, but, the accuracy is jumping from 0.93 and 1.0 and vice versa all the time.

```
X1 = 40*[0, 1, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0]
Y1 = 40*[0, 1, 0, 1, 1, 0, 1, 0, 1, 1, 1, 1]
```
